ExampleMod/Projectiles/ExampleSolarEruptionProjectile.cs

```diff
@@ -0,0 +1,212 @@
+		{
+			Player player = Main.player[projectile.owner];
+			
+			// Face the projectile towards its movement direction, offset by 90 degrees counterclockwise because the sprite faces downward.
+			projectile.rotation = projectile.velocity.ToRotation() + MathHelper.ToRadians(-90f);
+
+			// Constantly set the chain's timeLeft to 2 so that it doesn't die.
+			projectile.spriteDirection = projectile.direction;
+			projectile.timeLeft = 2;
+			player.ChangeDir(projectile.direction);
+			player.heldProj = projectile.whoAmI;
+			player.itemTime = 2;
+			player.itemAnimation = 2;
+			player.itemRotation = (projectile.velocity * projectile.direction).ToRotation();
+
+			// Makes some dust and light.
+			for (int i = 0; i < 3; i++)
+			{
+				Dust dust1 = Dust.NewDustPerfect(chainHeadPosition, DustID.Smoke, new Vector2(0f, -1f).RotatedByRandom(MathHelper.ToRadians(360f)), 150, Color.White, 1.5f);
+				dust1.noGravity = true;
+			}
+            Lighting.AddLight(chainHeadPosition, Color.White.ToVector3() * 0.4f);
+
+			// Increment the cooldown timer for spawning explosions.
+			if (projectile.localAI[1] > 0f)
+			{
+				projectile.localAI[1] -= 1f;
+			}
+
+			// The projectile's swerving motion.
+			if (projectile.localAI[0] == 0f)
+			{
+				projectile.localAI[0] = projectile.rotation;
+			}
+
+			float direction = (projectile.localAI[0].ToRotationVector2().X >= 0f).ToDirectionInt();
+			Vector2 rotation = (direction * (projectile.ai[0] / firingAnimation * MathHelper.ToRadians(360f) - MathHelper.ToRadians(90f))).ToRotationVector2();
+			rotation.Y *= (float)Math.Sin(projectile.ai[1]);
+
+			rotation = rotation.RotatedBy(projectile.localAI[0]);
+			projectile.ai[0] += 1f;
+			if (projectile.ai[0] < firingTime)
+			{
+				projectile.velocity += (firingSpeed * rotation).RotatedBy(MathHelper.ToRadians(90f));
+			}
+			else
+			{
+				projectile.Kill();
+			}
+
+			// Manages the positioning for the chain's handle.
+			Vector2 offset = Main.OffsetsPlayerOnhand[player.bodyFrame.Y / 56] * 2f;
+			if (player.direction == -1)
+			{
+				offset.X = player.bodyFrame.Width - offset.X;
+			}
+			if (player.gravDir == -1f)
+			{
+				offset.Y = player.bodyFrame.Height - offset.Y;
+			}
+			// This line is a custom offset that you can change to move the handle around. Default is 0f, 0f.
+			offset += new Vector2(4f, -6f) * new Vector2(player.direction, player.gravDir);
+			offset -= new Vector2(player.bodyFrame.Width - projectile.width, player.bodyFrame.Height - 42) * 0.5f;
+			projectile.Center = player.RotatedRelativePoint(player.position + offset) - projectile.velocity;
+		}
+
+		public override void OnHitNPC(NPC target, int damage, float knockback, bool crit)
+		{
+			// Spawns an explosion when it hits an NPC.
+			int cooldown = 3;
+			if (projectile.localAI[1] <= 0f)
+			{
+				Projectile explosion = Projectile.NewProjectileDirect(target.Center, Vector2.Zero, ModContent.ProjectileType<ExampleSolarEruptionExplosion>(), damage, 10f, projectile.owner, 0f, 1f);

```

Comment:
I honestly find a lot of this hard to follow
Splitting it up into local vars with comments would likely help a new-comer